File: api/agents.py
from http.server import BaseHTTPRequestHandler
import json

# List of available agents
AGENTS = [
    "MonologueAgent",
    "CodeActAgent", 
    "PlannerAgent"
]

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        self.send_response(200)
        self.send_header('Content-type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.end_headers()
        self.wfile.write(json.dumps(AGENTS).encode())
        return

    def do_OPTIONS(self):
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type, Authorization')
        self.end_headers()
        return

File: api/options/agents.ts
import type { VercelRequest, VercelResponse } from '@vercel/node';

/**
 * Vercel serverless function to serve available agents.
 * Returns a list of supported OpenHands agents.
 */
export default function handler(req: VercelRequest, res: VercelResponse) {
  // Set CORS headers
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  if (req.method !== 'GET') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  // Return the list of available agents
  const agents = [
    'CodeActAgent',
  ];

  return res.status(200).json(agents);
}
